builder/azure/arm/step_deploy_template.go

```diff
@@ -335,6 +350,8 @@ func (s *StepDeployTemplate) deleteDeploymentResources(ctx context.Context, subs
 					s.say(fmt.Sprintf("Couldn't delete %s resource. Will retry.\n"+
 						"Name: %s",
 						resourceType, resourceName))
+				} else {
+					s.say(fmt.Sprintf("Finished deleting -> %s : '%s'", resourceType, resourceName))
```

Comment:
In terms of UX, I wonder if it wouldn't be clearer to keep the successes/failures in a list and display them all in one message once the processing is done? This would reduce the amount of noise produced by the plugin during this step I would think

Reply:
I would prefer to not do that for now, with my refactored logic we'll just error saying that the VM, NIC, or resource failed, if the VM or NIC failed, the user will go manually delete it and see its associated resource, if we get an issue from users where this is confusing we can change this in the future